### PR TITLE
feat: jolt sdk

### PR DESCRIPTION
**Pull request by ncitron**
Initial implementation of jolt-sdk.

- Inputs and outputs are written to `examplename.joltdevice`
- Inputs are expected at the memory location `common::constants::INPUT_START_ADDRESS`
- Outputs are expected at memory location `common::constants::OUTPUT_START_ADDRESS`
- A 1 is written to `common::constants::PANIC_ADDRESS` if there was a panic at any point during the program execution
- Inputs are assumed to be initialized to 0 if not serialized into `examplename.joltdevice`
- Currently no way to feed inputs in using the compiler
    - Will address this in a follow up PR
    - This shouldn't be an issue since unset inputs are assumed to zero

### CODE REVIEW

File: common/src/constants.rs [hunk 1]
```diff
@@ -3,4 +3,8 @@ pub const REGISTER_COUNT: u64 = 32;
 pub const REGISTER_START_ADDRESS: usize = 0;
 pub const RAM_START_ADDRESS: u64 = 0x80000000;
 pub const BYTES_PER_INSTRUCTION: usize = 4;
-pub const MEMORY_OPS_PER_INSTRUCTION: usize = 7;
\ No newline at end of file
+pub const MEMORY_OPS_PER_INSTRUCTION: usize = 8;
```
Comment (ncitron) [inline common/src/constants.rs:6]: Did I change this or did this get modified recently?

**Comment by ncitron** [comment]
Also another follow up PR I need to look at here is to determine stable serialization of the inputs/outputs. Right now we use `mem::size_of` which has two problems. First, it will not work for pointer types (like `Vec<T>`) since those will just show up as the size of a pointer, which is wrong. Second, there is no guarantee that `size_of` is stable across compilations according to the docs. To address we should use some sort of serialization/deserialization scheme instead. We probably want one that is as efficient as possible. One choice that I think makes sense is bincode but open to suggestions.

File: common/src/constants.rs (hunk 1 above)
Comment (sragss) [reply common/src/constants.rs:6]: This should be 7. 

4x byte addressable, rd, rs1, rs2.

**Comment by moodlezoup** [comment]
squashed and incorporated into https://github.com/a16z/Lasso/pull/201

**Pull request closed without merging**